vpr/src/noc/odd_even_routing.cpp

```diff
@@ -35,9 +35,9 @@ const std::vector<TurnModelRouting::Direction>& OddEvenRouting::get_legal_direct
     const auto dst_router_pos = dst_router.get_router_physical_location();
 
     // get the compressed location for source, current, and destination NoC routers
-    auto compressed_src_loc = get_compressed_loc_approx(compressed_noc_grid,t_pl_loc{src_router_pos, 0}, num_layers)[src_router_pos.layer_num];
-    auto compressed_curr_loc = get_compressed_loc_approx(compressed_noc_grid,t_pl_loc{curr_router_pos, 0}, num_layers)[curr_router_pos.layer_num];
-    auto compressed_dst_loc = get_compressed_loc_approx(compressed_noc_grid,t_pl_loc{dst_router_pos, 0}, num_layers)[dst_router_pos.layer_num];
+    auto compressed_src_loc = get_compressed_loc_approx(compressed_noc_grid, t_pl_loc{src_router_pos, 0}, num_layers)[src_router_pos.layer_num];
```

Comment:
Check that you have commented why you use the compressed grid to check odd / even columns.